src/app/color_utils.cpp

```diff
@@ -37,12 +37,22 @@ gfx::Color color_utils::blackandwhite_neg(gfx::Color color)
     return gfx::rgba(0, 0, 0);
 }
 
-app::Color color_utils::color_from_ui(const gfx::Color color)
+app::Color color_utils::color_from_ui(const gfx::Color color, PixelFormat format)
```

Comment:
`gfx::Color` is always a 32-bit RGBA value. Here is like you are using a `gfx::Color` like a `doc::color_t`, probably you are looking for a new function, the inverse of `color_for_image()`.

Reply:
I'll review this later because the API is a little messy, mainly because the new kind of pixel format expected in `os::Paint`/SkPaint. From the docs of SkPaint:

    Sets alpha and RGB used when stroking and filling. The color is a 32-bit value,
    unpremultiplied, packing 8-bit components for alpha, red, blue, and green.

But it depends on the pixel format of the surface, so probably we need a new kind of color/type for these cases.

Reply:
> But it depends on the pixel format of the surface, so probably we need a new kind of color/type for these cases.

Yes, seems that one can set any unpremultiplied ARGB (8bit per component) color value to SkPaint and then the surface will just take what it needs depending on the SkColorType it has.

Reply:
Probably we can leave it as it is (because Skia itself expect the values in each channel of the RGBA value depending on the format), but I'd leave `color_from_ui()` as it was, and move the new implementation (the one with the `switch (format)`) inside `GraphicsContext_get_color()` itself, because it's the only place we need this (I guess).

Reply:
> but I'd leave `color_from_ui()` as it was, and move the new implementation (the one with the `switch (format)`) inside `GraphicsContext_get_color()` itself, because it's the only place we need this (I guess).

Sure, no problem, will make this change and see how it works. Thanks!